### PR TITLE
mysql-cluster 7.5.5

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Not sure if people aren't using this formula or just can't be bothered to update it.

### CODE REVIEW

**Comment by DomT4** [comment]
Pretty sure that's a random failure rather than anything of substance, hmm.

**Comment by JCount** [comment]
@DomT4 at first I thought it was probably random too, but I was able to reproduce it locally in a 10.11 VM and see the actual error.
The `bootstrap.log` produced:

```
/usr/local/Cellar/mysql-cluster/7.5.5/bin/mysqld --no-defaults --log-syslog=0 \
--bootstrap --basedir=/usr/local/Cellar/mysql-cluster/7.5.5 \
--datadir=/tmp/d20170219-24179-18ecowx/install.db --loose-skip-ndbcluster \
--tmpdir=/tmp/d20170219-24179-18ecowx/tmp/ \
--secure-file-priv=/tmp/d20170219-24179-18ecowx --innodb-log-file-size=5M --core-file \
--innodb_autoextend_increment=8 --innodb_buffer_pool_size=24M \
--lc-messages-dir=/usr/local/Cellar/mysql-cluster/7.5.5/share/mysql/ \
--character-sets-dir=/usr/local/Cellar/mysql-cluster/7.5.5/share/mysql/charsets \
--loose-auto_generate_certs=OFF --loose-sha256_password_auto_generate_rsa_keys=OFF
2017-02-19T23:51:17.635949Z 0 [Warning] --bootstrap is deprecated. \
Please consider using --initialize instead
dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
  Referenced from: /usr/local/Cellar/mysql-cluster/7.5.5/bin/mysqld
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _clock_gettime
  Referenced from: /usr/local/Cellar/mysql-cluster/7.5.5/bin/mysqld
  Expected in: /usr/lib/libSystem.B.dylib

safe_process[40450]: Child process: 40451, killed by signal: 5
```

**Comment by DomT4** [comment]
Oh, joy. I've tried a fix. We'll see how that goes I guess.

**Comment by DomT4** [comment]
Same error, I guess. These are all the mentions in the codebase:

```
client/dump/CMakeLists.txt
56:  IF(HAVE_clock_gettime_IN_rt)

configure.cmake
344:    MY_SEARCH_LIBS(clock_gettime rt LIBRT)
442:CHECK_FUNCTION_EXISTS (clock_gettime HAVE_CLOCK_GETTIME)
504:CHECK_FUNCTION_EXISTS (clock_gettime DNS_USE_CPU_CLOCK_FOR_ID)

libevent/configure.in
41:AC_CHECK_LIB(rt, clock_gettime)
139:AC_CHECK_FUNCS(gettimeofday vasprintf fcntl clock_gettime strtok_r strsep getaddrinfo getnameinfo strlcpy inet_ntop signal sigaction strtoll)
143:if test "x$ac_cv_func_clock_gettime" = "xyes"; then
144:   AC_DEFINE(DNS_USE_CPU_CLOCK_FOR_ID, 1, [Define if clock_gettime is available in libc])

libevent/evdns.c
1063:		if (clock_gettime(CLOCK_MONOTONIC, &ts) != -1)
1067:	if (clock_gettime(clkid, &ts) == -1)
1068:		event_err(1, "clock_gettime");

libevent/event.c
133:	if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
150:		if (clock_gettime(CLOCK_MONOTONIC, &ts) == -1)

libevent/WIN32-Code/event-config.h
11:/* Define if clock_gettime is available in libc */
17:/* Define to 1 if you have the `clock_gettime' function. */

mysys/my_getsystime.c
43:  clock_gettime(CLOCK_REALTIME, &tp);

mysys/my_rdtsc.c
64:#include <time.h>           /* for clock_gettime */
208:    clock_gettime(CLOCK_REALTIME, &tp);
816:   clock_gettime() -- In tests, clock_gettime often had

plugin/innodb_memcached/innodb_memcache/config.h
10:/* Define to 1 if you have the `clock_gettime' function. */

storage/ndb/memcache/src/schedulers/S_sched.cc
702: of calls to gettimeofday() or clock_gettime() to one per timeout_msec ms

storage/ndb/memcache/src/timing.c
60:  clock_gettime(CLOCK_MONOTONIC, &ts);
81:  return clock_gettime(PREFERRED_CLOCK, t);

storage/ndb/ndb_configure.cmake
49:CHECK_FUNCTION_EXISTS(clock_gettime HAVE_CLOCK_GETTIME)

storage/ndb/src/common/portlib/NdbCondition.c
57:  if ((res = clock_gettime(clock_id, &tick_time)) != 0)
173:  clock_gettime(clock_id, abstime);

storage/ndb/src/common/portlib/NdbMutex.c
177:  clock_gettime(CLOCK_MONOTONIC, &ts);

storage/ndb/src/common/portlib/NdbTick.cpp
54:  if (clock_gettime(NdbTick_clk_id, &tick_time) == 0)
58:  if (clock_gettime(NdbTick_clk_id, &tick_time) == 0)
67:  if (clock_gettime(NdbTick_clk_id, &tick_time) == 0)
70:  fprintf(stderr, "Failed to use CLOCK_REALTIME for clock_gettime,"
128:  const int res = clock_gettime(NdbTick_clk_id, &tick_time);
130:   * The only possible errors returned from clock_gettime()
143:    fprintf(stderr, "clock_gettime(%u, tp) failed, errno=%d\n",

storage/ndb/src/kernel/blocks/dbtup/DbtupDebug.cpp
56:  clock_gettime(CLOCK_REALTIME, tp);
63:  clock_gettime(CLOCK_REALTIME, &theStopTime);

storage/ndb/test/crund/helpers/src/utils/hrt_utils.c
42:    ret = clock_gettime(REAL_CLOCK_TYPE, &x->time);
76:    ret = clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &x->time);

storage/ndb/test/crund/helpers/src/utils/hrt_utils.h
41: * clock_gettime(), gettimeofday(), getrusage(), and times().
112: * Use: clock_gettime()		[SUSv2, POSIX.1-2001, #include <time.h>
```

**Comment by DomT4** [comment]
That seems to have worked, lol. CC @ilovezfs for this kind of fun.

**Comment by JCount** [comment]
🎉 @DomT4 good work! 💯

**Comment by MikeMcQuaid** [comment]
Ta again @DomT4 👍

File: Formula/mysql-cluster.rb [hunk 1]
```diff
@@ -38,10 +37,20 @@ class MysqlCluster < Formula
...
+    sha256 "727a932322d94287b62abb1bd2d41723eec4356a7728909e38adb65ca25241ca"
+  end
+
   def install
     # Make sure the var/mysql-cluster directory exists
     (var/"mysql-cluster").mkpath
 
+    # dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
+    if MacOS.version == "10.11" && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
+      inreplace "configure.cmake", "(clock_gettime", "(everything_is_terrible"
```
Comment (ilovezfs) [inline Formula/mysql-cluster.rb]: @DomT4 LOL everything_is_terrible :)

Did you have a chance to report the clockery upstream?

Comment (DomT4) [inline Formula/mysql-cluster.rb]: It felt appropriate 😉. I hadn't yet, no. To be honest if you went through my list of favourite bug reporting systems & how those bugs are handled MySQL stuff would be almost glued to the bottom 😕

**Pull request closed without merging**